# HW1 –Sentiment Analysis on Twitter 

###### Follow the instructionsin the file “Twitter HW Setup.pptx”to collect a good sample of tweets about a trending topic. You can find trending topics here.  (Suggestion: at least 500 tweets).Then, perform the following taskin a Jupyter notebook file.For each word, assign a value of +1 for positive sentiment, a value-1 for negative sentiment, or a value of 0 for neutral sentiment.You can easily find a list of “positive” words and a list of “negative” words online. Do you think that the general sentiment is negative or positive? Weakly or strongly?It is up to you to decidehow to measure the sentiment; make sure to briefly explain it in your submission though.

### Import Data

In [1]:
def importdata(filename):
    f = open(filename, 'r', encoding='utf8', errors='ignore')
    ds = []
    for line in f:
        line = line.split()
        for word in line:
            ds.append(word.lower().strip(',!?:;@'))
    f.close()
    return ds

In [2]:
def importdatadic(filename):
    f = open(filename, 'r', encoding='utf8', errors='ignore')
    ds = {}
    for line in f:
        line = line.split()
        for word in line:
            ds[word.lower().strip(',!?:;@')]=None
    f.close()
    return ds

### Count Words Frequentcy

In [3]:
def wordfreq(*t):
    j = 0
    d = {}
    for i in [t.count(w) for w in t]:
        d[t[j]]=i
        j=j+1
    d = sorted(d.items(), key=lambda x: x[1],reverse=True)
    return d

### Run Main Function

In [10]:
def main():
    # Import Data
    txt = importdata('Spider-Man.txt')
    negative = importdatadic('negative-words.txt')
    positive = importdatadic('positive-words.txt')
    scr = 0
    w = 0
    ptxt = []
    ntxt = []
    for word in txt:
        if word in positive:
            scr = scr + 1
            ptxt.append(word)
        if word in negative:
            scr = scr - 1
            ntxt.append(word)
        w = w + 1
    print(scr) 
    print(str(scr / w *100) + '%')
    # List the top 10 words
    pd = wordfreq(*ptxt)
    nd = wordfreq(*ntxt)
    print(pd[0:9])
    print(nd[0:9])
       
if __name__ == "__main__": 
    # calling main function 
    main()

3340
1.0157100543131186%
[('like', 919), ('marvel', 782), ('good', 605), ('liked', 603), ('protect', 530), ('amazing', 489), ('best', 453), ('enough', 452), ('fans', 344)]
[('sinister', 596), ('fucking', 593), ('complex', 530), ('cry', 368), ('parody', 328), ('stark', 280), ('fury', 243), ('shit', 163), ('missed', 137)]


##### The score of Spider-Man is 3340 which indicates that the general sentiment on Twitter is positive. 
However, there are some concern about the scoring method. 
Fisrt of all, the most common positive words show directly positive emotions. But on the other side, the most common negative words are more complex and hard to tell the true opinion by single words. For example, 'fucking', 'complex', 'cry' and 'missed' are words that could not reflex the meaning directly.
Second, some of the most common positive or negative words, 'marvel', 'sinister', 'stark', might not indicates positive or negative meaning. Instead, they are just the name of the characters or relavent nouns.